### DISCUSSION TITLE
(Pre-)Configuring pocketbase for certain setups

### DISCUSSION THREAD

**Discussion by AliSajid** [pocketbase/pocketbase#944] [Q&A]
Hi,

I am playing with pocketbase (Thank you Fireship.io) and running a small Application that I'm building as a proof of concept. My current setup is to use a docker image to run it off fly.io. As I am configuring it, I have to run the same configurations (Adding admins and users, setting up the SMTP Stuff, and file storage, etc).

My question is:

* Is there a way to do this automatically via a config file of sorts OR
* Should I create a small script that runs during the docker image build and uses the API for creating these things.

**Comment by ganigeorgiev** [comment]
Pre-configurations via code currently can be done if you use [PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) either by updating `app.Settings()` in the [`OnBeforeServe` hook](https://pocketbase.io/docs/event-hooks/) OR via [DB migrations](http://localhost:5173/docs/db-migrations).

Here is an example using the hook:

```go
package main

import (
    "log"

    "github.com/pocketbase/pocketbase"
    "github.com/pocketbase/pocketbase/core"
)

func main() {
    app := pocketbase.New()

    app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
        // note: This will overwrite any previously loaded app settings value in place.
        //       You may want to add some checks when to apply it or use a db migration
        //       so that it can run only once.
        app.Settings().Merge(&core.Settings{
            Smtp: core.SmtpConfig{
                Enabled:  false,
                Host:     "smtp.example.com",
                Port:     587,
                Username: "",
                Password: "",
                Tls:      false,
            },
        })

        return nil
    })

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

More documentation on this will be available in the next 1-2 weeks with the v0.8.0 release.

In the future (v0.9/v0.10) there are plans to "extend" the Go api with JS hooks for the cases when the prebuilt binary is used.

**Reply by ganigeorgiev** [reply]
Alternatively, you can also configure once your settings from the Admin UI and then make a copy of the `pb_data/data.db` when initializing the project.

**Reply by AliSajid** [reply]
Thank you. This is wonderful. I will likely go with the `onBeforeServe` option and see where it takes me. Will keep you updated.
